client_metrics.go

```diff
@@ -127,50 +131,54 @@ func (m *ClientMetrics) Collect(ch chan<- prom.Metric) {
 // EnableClientHandlingTimeHistogram turns on recording of handling time of RPCs.
 // Histogram metrics can be very expensive for Prometheus to retain and query.
 func (m *ClientMetrics) EnableClientHandlingTimeHistogram(opts ...HistogramOption) {
-	for _, o := range opts {
-		o(&m.clientHandledHistogramOpts)
-	}
-	if !m.clientHandledHistogramEnabled {
-		m.clientHandledHistogram = prom.NewHistogramVec(
-			m.clientHandledHistogramOpts,
-			[]string{"grpc_type", "grpc_service", "grpc_method"},
-		)
-	}
-	m.clientHandledHistogramEnabled = true
+	m.doOnceClientHandledHistogramEnable.Do(func() {
+		for _, o := range opts {
+			o(&m.clientHandledHistogramOpts)
+		}
+		if !m.clientHandledHistogramEnabled {
+			m.clientHandledHistogram = prom.NewHistogramVec(
+				m.clientHandledHistogramOpts,
+				[]string{"grpc_type", "grpc_service", "grpc_method"},
+			)
+			m.clientHandledHistogramEnabled = true
```

Comment:
I moved this up into the `if` block, since there's no point in overwriting `true` with `true`...